File: fix_jsx_syntax.js
const fs = require('fs');
const path = require('path');

// Function to fix JSX syntax errors
function fixJSXSyntax(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Fix common JSX syntax issues
    const fixes = [
      // Fix missing opening braces in function declarations
      { pattern: /const\s+(\w+)\s*=\s*\(\)\s*=>\s*\{\}\s*\n\s*return\s*\(\)/g, replacement: 'const $1 = () => {\n  return (' },
      { pattern: /const\s+(\w+)\s*=\s*\(\)\s*=>\s*\{\}\s*\n\s*return\s*\(/g, replacement: 'const $1 = () => {\n  return (' },
      
      // Fix missing opening braces in arrow functions
      { pattern: /\(\s*\)\s*=>\s*\{\}\s*\n\s*return\s*\(\)/g, replacement: '() => {\n  return (' },
      { pattern: /\(\s*\)\s*=>\s*\{\}\s*\n\s*return\s*\(/g, replacement: '() => {\n  return (' },
      
      // Fix missing opening parentheses in return statements
      { pattern: /return\s*\(\)\s*\n/g, replacement: 'return (\n' },
      
      // Fix malformed JSX closing tags
      { pattern: /<\/div>\s*<\/div>\s*<\/div>/g, replacement: '</div>' },
      { pattern: /<\/div>\s*<\/div>/g, replacement: '</div>' },
      
      // Fix malformed JSX opening tags
      { pattern: /<div\s*className="[^"]*">\s*<\/div>\s*<\/div>/g, replacement: '<div className="$1">' },
      
      // Fix missing closing braces and parentheses
      { pattern: /return\s*\(\s*<[^>]*>\s*$/gm, replacement: 'return (\n    <div>' },
      
      // Fix malformed JSX expressions
      { pattern: /\{\s*\}\s*<\/div>/g, replacement: '</div>' },
      { pattern: /\{\s*\}\s*<\/span>/g, replacement: '</span>' },
      { pattern: /\{\s*\}\s*<\/p>/g, replacement: '</p>' },
      { pattern: /\{\s*\}\s*<\/h1>/g, replacement: '</h1>' },
      { pattern: /\{\s*\}\s*<\/h2>/g, replacement: '</h2>' },
      { pattern: /\{\s*\}\s*<\/h3>/g, replacement: '</h3>' },
      { pattern: /\{\s*\}\s*<\/button>/g, replacement: '</button>' },
      { pattern: /\{\s*\}\s*<\/a>/g, replacement: '</a>' },
      
      // Fix malformed JSX content
      { pattern: /<h1[^>]*>\s*<\/div>\s*<\/div>\s*([^<]+)\s*<\/h1>/g, replacement: '<h1$1>$2</h1>' },
      { pattern: /<h2[^>]*>\s*<\/div>\s*<\/div>\s*([^<]+)\s*<\/h2>/g, replacement: '<h2$1>$2</h2>' },
      { pattern: /<h3[^>]*>\s*<\/div>\s*<\/div>\s*([^<]+)\s*<\/h3>/g, replacement: '<h3$1>$2</h3>' },
      { pattern: /<p[^>]*>\s*<\/div>\s*<\/div>\s*([^<]+)\s*<\/p>/g, replacement: '<p$1>$2</p>' },
      
      // Fix malformed div structures
      { pattern: /<div[^>]*>\s*<\/div>\s*<\/div>\s*<div[^>]*>/g, replacement: '<div$1><div$2>' },
      
      // Fix missing semicolons
      { pattern: /(\w+)\s*=\s*\([^)]*\)\s*=>\s*\{[^}]*\}\s*(?=\n\s*const|\n\s*export|\n\s*import|\n\s*function|\n\s*class|\n\s*interface|\n\s*type|\n\s*$)/g, replacement: '$1 = ($2) => {\n  $3\n};' },
      
      // Fix missing commas in object literals
      { pattern: /(\w+):\s*([^,}]+)\s*(\w+):/g, replacement: '$,
  1: $2,\n  $3:' },
      
      // Fix missing semicolons in variable declarations
      { pattern: /const\s+(\w+)\s*=\s*[^;]+(?=\n\s*const|\n\s*export|\n\s*import|\n\s*function|\n\s*class|\n\s*interface|\n\s*type|\n\s*$)/g, replacement: 'const $1 = $2;' },
      { pattern: /let\s+(\w+)\s*=\s*[^;]+(?=\n\s*const|\n\s*export|\n\s*import|\n\s*function|\n\s*class|\n\s*interface|\n\s*type|\n\s*$)/g, replacement: 'let $1 = $2;' },
      { pattern: /var\s+(\w+)\s*=\s*[^;]+(?=\n\s*const|\n\s*export|\n\s*import|\n\s*function|\n\s*class|\n\s*interface|\n\s*type|\n\s*$)/g, replacement: 'var $1 = $2;' },
    ];
    
    for (const fix of fixes) {
      const newContent = content.replace(fix.pattern, fix.replacement);
      if (newContent !== content) {
        content = newContent;
        modified = true;
      }
    }
    
    if (modified) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed JSX syntax in: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Function to recursively fix files
function fixAllJSXSyntax(dir) {
  const files = fs.readdirSync(dir);
  let fixedCount = 0;
  
  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      // Skip node_modules and .git directories
      if (file === 'node_modules' || file === '.git' || file === '.next') {
        continue;
      }
      fixedCount += fixAllJSXSyntax(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts') || file.endsWith('.js') || file.endsWith('.jsx')) {
      if (fixJSXSyntax(filePath)) {
        fixedCount++;
      }
    }
  }
  
  return fixedCount;
}

// Main execution
console.log('Starting JSX syntax fixes...');
const fixedCount = fixAllJSXSyntax('.');
console.log(`Fixed JSX syntax in ${fixedCount} files.`);

File: fix_jsx_syntax.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_jsx_syntax(file_path):
    """Fix JSX syntax errors in TypeScript/JavaScript files"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix common JSX syntax issues
        # 1. Fix missing React import
        if 'import React' not in content and ('<div>' in content or '<>' in content or '<Head>' in content):
            content = "import React from 'react';\n" + content
        
        # 2. Fix missing Head import
        if '<Head>' in content and 'import Head' not in content:
            content = content.replace("import React from 'react';", "import React from 'react';\nimport Head from 'next/head';")
        
        # 3. Fix malformed JSX return statements
        content = re.sub(r'return\s*\(\s*<div>', 'return (\n    <div>', content)
        content = re.sub(r'return\s*\(\s*<>', 'return (\n    <>', content)
        
        # 4. Fix malformed meta tags
        content = re.sub(r'<meta name = "([^"]+)"', '<meta name="\1"', content)
        
        # 5. Fix missing closing tags
        content = re.sub(r'<Head>\s*<title>([^<]+)</title>\s*<meta[^>]+>\s*</Head>', 
                        '<Head>\n        <title>\1</title>\n        <meta name="description" content="Professional services by Zion Tech Group." />\n      </Head>', content)
        
        # 6. Fix malformed function declarations
        content = re.sub(r'export default function (\w+)\(\) \{', r'export default function \1() {\n  return (', content)
        
        # 7. Fix missing closing parentheses and braces
        if 'return (' in content and not content.strip().endswith(');'):
            content = content.rstrip() + '\n  );\n}'
        
        # 8. Fix malformed JSX fragments
        content = re.sub(r'<>\s*<div>', '<>\n      <div>', content)
        content = re.sub(r'</div>\s*</>', '</div>\n    </>', content)
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        
        return False
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all TypeScript and JavaScript files
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts', 
        'app/**/*.js',
        'app/**/*.jsx'
    ]
    
    files_fixed = 0
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if fix_jsx_syntax(file_path):
                files_fixed += 1
                print(f"Fixed JSX syntax in: {file_path}")
    
    print(f"Fixed JSX syntax in {files_fixed} files")

if __name__ == "__main__":
    main()
